components/infobox/wikis/smite/infobox_person_player_custom.lua

```diff
@@ -34,20 +66,72 @@ function CustomPlayer.run(frame)
 		iconModule = 'Module:PositionIcon/data',
 		player = player.pagename
 	}
-	player.role = Role.run{role = player.args.role}
-	player.role2 = Role.run{role = player.args.role2}
+	player.role = player:_getRoleData(player.args.role)
+	player.role2 = player:_getRoleData(player.args.role2)
 
 	return player:createInfobox()
 end
 
 function CustomInjector:parse(id, widgets)
-	if id == 'region' then return {}
-	elseif id == 'role' then
+	local caller = self.caller
+
+	if id == 'role' then
 		return {
-			Cell{name = 'Role(s)', content = {self.caller.role.display, self.caller.role2.display}}
+			Cell{name = 'Role', content = {
+				caller:_displayRole(caller.role),
+				caller:_displayRole(caller.role2),
+			}},
 		}
 	end
 	return widgets
 end
 
+---@param role string?
+---@return {category: string, variable: string, isplayer: boolean?}?
+function CustomPlayer:_getRoleData(role)
+	return ROLES[(role or ''):lower()]
+end
+
+---@param roleData {category: string, variable: string, isplayer: boolean?}?
+---@return string?
+function CustomPlayer:_displayRole(roleData)
+	if not roleData then return end
+
+	return Page.makeInternalLink(roleData.variable, ':Category:' .. roleData.category)
+end
+
+---@param categories string[]
+---@return string[]
+function CustomPlayer:getWikiCategories(categories)
+	return Array.append(categories,
+		(self.role or {}).category,
+		(self.role2 or {}).category
+	)
+end
+
+---@param lpdbData table
+---@param args table
+---@param personType string
+---@return table
+function CustomPlayer:adjustLPDB(lpdbData, args, personType)
+	lpdbData.extradata.role = (self.role or {}).variable
+	lpdbData.extradata.role2 = (self.role2 or {}).variable
+	lpdbData.type = self:_isPlayerOrStaff()
+	return lpdbData
+end
+
+---@return string
+function CustomPlayer:_isPlayerOrStaff()
+	local roleData
+	if String.isNotEmpty(self.args.role) then
+		roleData = ROLES[self.args.role:lower()]
+	end
+	-- If the role is missing, assume it is a player
+	if roleData and roleData.isplayer == false then
+		return 'staff'
+	else
+		return 'player'
+	end
+end
```

Comment:
Same

Reply:
Is there a way where the roles could determine player type? 

currently after I removed this, while its out of scope: The player portal counts coach as player type (and did not cause a separate section), it was actually why I have this added in the 1st rework revision

![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/8e06a74f-d5dd-4ed0-9e36-d9fb44219d50)

Reply:
```lua
---@param args table
---@return {store: string, category: string}
function CustomPlayer:getPersonType(args)
	......
end
```

Reply:
> ```lua
> ---@param args table
> ---@return {store: string, category: string}
> function CustomPlayer:getPersonType(args)
> 	......
> end
> ```

but this would mean I need to bring back the role storage in extradata and isplayer back then to make this work? or there's a setup that can make it work without requiring both?